test/example/small_set.cpp

```diff
@@ -0,0 +1,178 @@
+#include <algorithm>
+#include <cassert>
+#include <iostream>
+#include <vector>
+
+#include "support/small_set.h"
+
+using namespace wasm;
+
+template<typename T>
+void assertContents(T& t, const std::vector<int>& expectedContents) {
+  assert(t.size() == expectedContents.size());
+  for (auto item : expectedContents) {
+    assert(t.count(item) == 1);
+  }
+  // Also test this using an iterator and a const iterator to also get
+  // coverage there.
+  for (auto item : t) {
```

Comment:
This would fail to compile atm since I couldn't figure out how to get the non-const iterator to return a reference, https://github.com/WebAssembly/binaryen/pull/4188#discussion_r717032029

(As nothing needs a reference atm, I suggest that shouldn't block this from landing, and leave it for a TODO)